xarray/core/variable.py

```diff
@@ -893,16 +893,27 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
         if dim is not None and axis is not None:
             raise ValueError("cannot supply both 'axis' and 'dim' arguments")
 
+        if 'cum' in func.__name__:
```

Comment:
I didn't individual test dataarray and dataset functionality but assume it is inherited by variable.  Hence, I've tested variable methods for cumsum and cumprod for now.